client/checkpoint_create_test.go

```diff
@@ -0,0 +1,55 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+func TestCheckpointCreateError(t *testing.T) {
+	client := &APIClient{
+		HTTPCli: newMockClient(errorMockResponse(http.StatusInternalServerError, "Server error")),
+	}
+	err := client.ContainerCheckpointCreate(context.Background(), "nothing", &types.CheckpointCreateOptions{CheckpointID: "cp0"})
+	if err == nil || !strings.Contains(err.Error(), "Server error") {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+}
+
+func TestCheckpointCreate(t *testing.T) {
+	expectedURL := "/containers/id/checkpoints"
+
+	httpClient := newMockClient(func(req *http.Request) (*http.Response, error) {
+		if !strings.HasPrefix(req.URL.Path, expectedURL) {
+			return nil, fmt.Errorf("expected URL '%s', got '%s'", expectedURL, req.URL)
+		}
+		if req.Header.Get("Content-Type") == "application/json" {
```

Comment:
if the `Content-Type` doesn't equal `application/json`, the case should fail, right?

Reply:
I re-think this problem, maybe it also can accepted non-json type? @fuweid